bitnami/mlflow/templates/tracking/deployment.yaml

```diff
@@ -185,8 +185,6 @@ spec:
                   name: {{ include "mlflow.v0.s3.secretName" . }}
                   key: {{ include "mlflow.v0.s3.secretAccessKeyKey" . | quote }}
             {{- end }}
-            - name: MLFLOW_S3_ENDPOINT_URL
-              value: {{ printf "%s://%s:%v" (include "mlflow.v0.s3.protocol" .) (include "mlflow.v0.s3.host" .) (include "mlflow.v0.s3.port" .) | quote }}
```

Comment:
This would change the default behavior of the chart since it is needed when minio is enabled, based on [this comment|https://github.com/bitnami/charts/issues/23959#issuecomment-2032686094]

Reply:
Thanks for raising this! @iamhritik, I suggested that there would be no problem on removing the env-var as users could still use `tracking.extraEnvVars` to add it for specific scenarios requiring it: https://github.com/bitnami/charts/issues/23959#issuecomment-2060752106, but it turns out that the default architecture of this chart is precisely deploying MinIO. So we will need it. What if we wrap the env-var in an if condition?

```
            {{- if .Values.minio.enabled }}
            - name: MLFLOW_S3_ENDPOINT_URL
              value: {{ printf "%s://%s:%v" (include "mlflow.v0.s3.protocol" .) (include "mlflow.v0.s3.host" .) (include "mlflow.v0.s3.port" .) | quote }}
            {{- end }}
```

Reply:
Got it! @andresbono I'll proceed with updating the PR.

Reply:
updated tracking deployment code with this new logic.